sdk/tables/Azure.Data.Tables/src/TableClientOptions.cs

```diff
@@ -30,10 +30,17 @@ public TableClientOptions(ServiceVersion serviceVersion = Latest)
             VersionString = serviceVersion switch
             {
                 ServiceVersion.V2019_02_02 => "2019-02-02",
+                ServiceVersion.V2020_12_06 => "2020-12-06",
                 _ => throw new ArgumentOutOfRangeException(nameof(serviceVersion))
             };
         }
 
+        /// <summary>
+        ///  Enables tenant discovery through the authorization challenge when the client is configured to use a TokenCredential.
+        /// When <c>true</c>, the client will attempt an initial un-authorized request to prompt an OAuth challenge in order to discover the correct tenant for the resource.
+        /// </summary>
+        public bool EnableTenantDiscovery { get; set; }
```

Comment:
Why would someone use this?

Reply:
In scenarios where the user is a member of many tenants, and the tenant of the specific table resource isn't necessarily known at compile time. This is already an established pattern for Key Vault and Storage.